From b02c2766d1fee7f627a138f62acbcb31733cd6e1 Mon Sep 17 00:00:00 2001
From: "tao.zeng" <tao.zeng@amlogic.com>
Date: Fri, 16 May 2014 18:42:42 +0800
Subject: [PATCH 4120/5965] [M8B] driver change for dvfs and pmu

1. update dvfs table for all m8b project;
2. Increase max cpu frequent to 1608MHz;
3. restart charger when detect fault of charger.
---
 .../boot/dts/amlogic/meson8b_m101_512M.dtd    |  27 ++---
 arch/arm/boot/dts/amlogic/meson8b_m102_1G.dtd |  27 ++---
 arch/arm/boot/dts/amlogic/meson8b_m200_1G.dtd |  20 ++--
 arch/arm/boot/dts/amlogic/meson8b_m201_1G.dtd | 110 ++----------------
 arch/arm/mach-meson8b/clock.c                 |   2 +-
 drivers/amlogic/power/aml_pmu/aml1218-sply.c  |   4 +-
 6 files changed, 47 insertions(+), 143 deletions(-)

diff --git a/arch/arm/boot/dts/amlogic/meson8b_m101_512M.dtd b/arch/arm/boot/dts/amlogic/meson8b_m101_512M.dtd
index 9f6d8a794e05..fb3bb7aa01e7 100755
--- a/arch/arm/boot/dts/amlogic/meson8b_m101_512M.dtd
+++ b/arch/arm/boot/dts/amlogic/meson8b_m101_512M.dtd
@@ -701,21 +701,22 @@ sdhc{
 //$$ L2 PROP_U32 16*3 = "dvfs_table"
         vcck_dvfs {
             dvfs_id     = <1>;                                  /** must be value of (1 << n)            */
-            table_count = <12>;                                 /** must be correct count for dvfs_table */
+            table_count = <13>;                                 /** must be correct count for dvfs_table */
             dvfs_table  = <
             /* NOTE: frequent in this table must be ascending order */
             /* frequent(Khz)    min_uV      max_uV                  */
-                  96000         875000      875000
-                 192000         875000      875000
-                 312000         875000      875000
-                 408000         875000      875000
-                 504000         875000      875000
-                 600000         900000      900000
-                 720000         900000      900000
-                 816000         925000      925000
-                1008000         975000      975000
-                1200000        1025000     1025000
-                1416000        1075000     1075000
+                  96000         825000      825000
+                 192000         825000      825000
+                 312000         825000      825000
+                 408000         825000      825000
+                 504000         825000      825000
+                 600000         825000      825000
+                 720000         825000      825000
+                 816000         850000      850000
+                1008000         875000      875000
+                1200000         925000      925000
+                1320000        1000000     1000000
+                1488000        1075000     1075000
                 1608000        1150000     1150000
             >;
         };
@@ -773,7 +774,7 @@ sdhc{
         status = "okay";
 	    pmu_twi_id = <2>;                   /** PMU i2c bus                          */
         pmu_irq_id = <0>;                   /** INT_WATCHDOG                         */
-        pmu_twi_addr = <0x32>;              /** PMU i2c addr                         */
+        pmu_twi_addr = <0x35>;              /** PMU i2c addr                         */
         pmu_battery_rdc = <100>;            /** battery internal resistor            */
         pmu_battery_cap = <3670>;           /** battery capacity                     */
         pmu_battery_technology = <4>;       /** POWER_SUPPLY_TECHNOLOGY_LiFe         */
diff --git a/arch/arm/boot/dts/amlogic/meson8b_m102_1G.dtd b/arch/arm/boot/dts/amlogic/meson8b_m102_1G.dtd
index 38c9619c8527..96a4d8cdd09a 100755
--- a/arch/arm/boot/dts/amlogic/meson8b_m102_1G.dtd
+++ b/arch/arm/boot/dts/amlogic/meson8b_m102_1G.dtd
@@ -680,21 +680,22 @@ void root_func(){
 //$$ L2 PROP_U32 16*3 = "dvfs_table"
         vcck_dvfs {
             dvfs_id     = <1>;                                  /** must be value of (1 << n)            */
-            table_count = <12>;                                 /** must be correct count for dvfs_table */
+            table_count = <13>;                                 /** must be correct count for dvfs_table */
             dvfs_table  = <
             /* NOTE: frequent in this table must be ascending order */
             /* frequent(Khz)    min_uV      max_uV                  */
-                  96000         875000      875000
-                 192000         875000      875000
-                 312000         875000      875000
-                 408000         875000      875000
-                 504000         875000      875000
-                 600000         900000      900000
-                 720000         900000      900000
-                 816000         925000      925000
-                1008000         975000      975000
-                1200000        1025000     1025000
-                1416000        1075000     1075000
+                  96000         825000      825000
+                 192000         825000      825000
+                 312000         825000      825000
+                 408000         825000      825000
+                 504000         825000      825000
+                 600000         825000      825000
+                 720000         825000      825000
+                 816000         850000      850000
+                1008000         875000      875000
+                1200000         925000      925000
+                1320000        1000000     1000000
+                1488000        1075000     1075000
                 1608000        1150000     1150000
             >;
         };
@@ -752,7 +753,7 @@ void root_func(){
         status = "okay";
 	    pmu_twi_id = <2>;                   /** PMU i2c bus                          */
         pmu_irq_id = <0>;                   /** INT_WATCHDOG                         */
-        pmu_twi_addr = <0x32>;              /** PMU i2c addr                         */
+        pmu_twi_addr = <0x35>;              /** PMU i2c addr                         */
         pmu_battery_rdc = <100>;            /** battery internal resistor            */
         pmu_battery_cap = <3670>;           /** battery capacity                     */
         pmu_battery_technology = <4>;       /** POWER_SUPPLY_TECHNOLOGY_LiFe         */
diff --git a/arch/arm/boot/dts/amlogic/meson8b_m200_1G.dtd b/arch/arm/boot/dts/amlogic/meson8b_m200_1G.dtd
index 16f427ea4bfd..5b0bf40509ce 100755
--- a/arch/arm/boot/dts/amlogic/meson8b_m200_1G.dtd
+++ b/arch/arm/boot/dts/amlogic/meson8b_m200_1G.dtd
@@ -668,28 +668,24 @@ void root_func(){
 //$$ L2 PROP_U32 16*3 = "dvfs_table"
         vcck_dvfs {
             dvfs_id     = <1>;                                  /** must be value of (1 << n)            */
-            table_count = <17>;                                 /** must be correct count for dvfs_table */
+            table_count = <13>;                                 /** must be correct count for dvfs_table */
             change-frequent-only;
             dvfs_table  = <
             /* NOTE: frequent in this table must be ascending order */
             /* frequent(Khz)    min_uV      max_uV                  */
-                  24000         825000      825000
                   96000         825000      825000
                  192000         825000      825000
                  312000         825000      825000
                  408000         825000      825000
                  504000         825000      825000
-                 600000         850000      850000
-                 720000         850000      850000
-                 816000         875000      875000
-                 912000         900000      900000
-                1008000         925000      925000
-                1104000         950000      950000
-                1200000         975000      975000
+                 600000         825000      825000
+                 720000         825000      825000
+                 816000         850000      850000
+                1008000         875000      875000
+                1200000         925000      925000
                 1320000        1000000     1000000
-                1416000        1025000     1025000
-                1512000        1050000     1050000
-                1608000        1100000     1100000
+                1488000        1075000     1075000
+                1608000        1150000     1150000
             >;
         };
     };
diff --git a/arch/arm/boot/dts/amlogic/meson8b_m201_1G.dtd b/arch/arm/boot/dts/amlogic/meson8b_m201_1G.dtd
index c5349db94770..e83b1c84c841 100755
--- a/arch/arm/boot/dts/amlogic/meson8b_m201_1G.dtd
+++ b/arch/arm/boot/dts/amlogic/meson8b_m201_1G.dtd
@@ -700,28 +700,24 @@ void root_func(){
 //$$ L2 PROP_U32 16*3 = "dvfs_table"
         vcck_dvfs {
             dvfs_id     = <1>;                                  /** must be value of (1 << n)            */
-            table_count = <17>;                                 /** must be correct count for dvfs_table */
+            table_count = <13>;                                 /** must be correct count for dvfs_table */
             change-frequent-only;
             dvfs_table  = <
             /* NOTE: frequent in this table must be ascending order */
             /* frequent(Khz)    min_uV      max_uV                  */
-                  24000         825000      825000
                   96000         825000      825000
                  192000         825000      825000
                  312000         825000      825000
                  408000         825000      825000
                  504000         825000      825000
-                 600000         850000      850000
-                 720000         850000      850000
-                 816000         875000      875000
-                 912000         900000      900000
-                1008000         925000      925000
-                1104000         950000      950000
-                1200000         975000      975000
+                 600000         825000      825000
+                 720000         825000      825000
+                 816000         850000      850000
+                1008000         875000      875000
+                1200000         925000      925000
                 1320000        1000000     1000000
-                1416000        1025000     1025000
-                1512000        1050000     1050000
-                1608000        1100000     1100000
+                1488000        1075000     1075000
+                1608000        1150000     1150000
             >;
         };
     };
@@ -738,96 +734,6 @@ void root_func(){
                       0     154     0x04>;
     };
 
-//$$ DEVICE="Battery_parameter"
-//$$ MATCH "battery_parameter_match" = <&battery_parameter>
-//$$ L2 PROP_STR = "status"
-//$$ L2 PROP_U32 = "pmu_twi_id"
-//$$ L2 PROP_U32 = "pmu_irq_id"
-//$$ L2 PROP_U32 = "pmu_twi_addr"
-//$$ L2 PROP_U32 = "pmu_battery_rdc"
-//$$ L2 PROP_U32 = "pmu_battery_cap"
-//$$ L2 PROP_U32 = "pmu_battery_technology"
-//$$ L2 PROP_STR = "pmu_battery_name"
-//$$ L2 PROP_U32 = "pmu_init_chgvol"
-//$$ L2 PROP_U32 = "pmu_init_chgend_rate"
-//$$ L2 PROP_U32 = "pmu_init_chg_enabled"
-//$$ L2 PROP_U32 = "pmu_init_adc_freq"
-//$$ L2 PROP_U32 = "pmu_init_adc_freqc"
-//$$ L2 PROP_U32 = "pmu_init_chg_pretime"
-//$$ L2 PROP_U32 = "pmu_init_chg_csttime"
-//$$ L2 PROP_U32 = "pmu_init_chgcur"
-//$$ L2 PROP_U32 = "pmu_suspend_chgcur"
-//$$ L2 PROP_U32 = "pmu_resume_chgcur"
-//$$ L2 PROP_U32 = "pmu_shutdown_chgcur"
-//$$ L2 PROP_U32 = "pmu_usbcur_limit"
-//$$ L2 PROP_U32 = "pmu_usbcur"
-//$$ L2 PROP_U32 = "pmu_usbvol_limit"
-//$$ L2 PROP_U32 = "pmu_usbvol"
-//$$ L2 PROP_U32 = "pmu_pwroff_vol"
-//$$ L2 PROP_U32 = "pmu_pwron_vol"
-//$$ L2 PROP_U32 = "pmu_pekoff_time"
-//$$ L2 PROP_U32 = "pmu_pekoff_en"
-//$$ L2 PROP_U32 = "pmu_peklong_time"
-//$$ L2 PROP_U32 = "pmu_pwrok_time"
-//$$ L2 PROP_U32 = "pmu_pwrnoe_time"
-//$$ L2 PROP_U32 = "pmu_intotp_en"
-//$$ L2 PROP_U32 = "pmu_pekon_time"
-//$$ L2 PROP_U32 = "pmu_charge_efficiency"
-//$$ L2 PROP_U32 16*3 ="pmu_bat_curve"
-    battery_parameter:battery_parameter{
-        status = "okay";
-        pmu_twi_id = <2>;                   /** PMU i2c bus                          */
-        pmu_irq_id = <0>;                   /** INT_WATCHDOG                         */
-        pmu_twi_addr = <0x32>;              /** PMU i2c addr                         */
-        pmu_battery_rdc = <94>;             /** battery internal resistor            */
-        pmu_battery_cap = <9250>;           /** battery capacity                     */
-        pmu_battery_technology = <4>;       /** POWER_SUPPLY_TECHNOLOGY_LiFe         */
-        pmu_battery_name = "PTI PL336078";  /** battery name                         */
-        pmu_init_chgvol = <4200000>;        /** target charge voltage                */
-        pmu_init_chgend_rate = <10>;        /** set initial charing end current rate */
-        pmu_init_chg_enabled = <1>;         /** set initial charing enabled          */
-        pmu_init_adc_freq = <25>;           /** set initial adc frequency            */
-        pmu_init_adc_freqc = <100>;         /** set initial coulomb adc coufrequency */
-        pmu_init_chg_pretime = <80>;        /** set initial pre-charging time        */
-        pmu_init_chg_csttime = <300>;       /** set initial rapid-charging time      */
-        pmu_init_chgcur = <300000>;         /** set initial charging current limite  */
-        pmu_suspend_chgcur = <1300000>;     /** set suspend charging current limite  */
-        pmu_resume_chgcur = <300000>;       /** set resume charging current limite   */
-        pmu_shutdown_chgcur = <1300000>;    /** set shutdown charging current limite */
-        pmu_usbcur_limit = <1>;             /** enable usb current limit?            */
-        pmu_usbcur = <500>;                 /** usb limit current value if enabled   */
-        pmu_usbvol_limit = <1>;             /** enable usb vbus voltage limit        */
-        pmu_usbvol = <4400>;                /** voltage limit value if enabled       */
-        pmu_pwroff_vol = <3200>;            /** hardware force power off voltage     */
-        pmu_pwron_vol = <2900>;             /** power on when battery voltage above  */
-        pmu_pekoff_time = <6000>;           /** long press to power off system       */
-        pmu_pekoff_en  = <1>;               /** enable long press to PWR OFF system  */
-        pmu_peklong_time = <1500>;          /** threshold to detect long press       */
-        pmu_pwrok_time   = <64>;            /** delay to detect pwr ok signal        */
-        pmu_pwrnoe_time = <2000>;           /**                                      */
-        pmu_intotp_en = <1>;                /** over temperature protection enable   */
-        pmu_pekon_time = <128>;             /** powerkey hold time for power on      */
-        pmu_charge_efficiency = <99>;       /** efficient of charging                */
-        pmu_bat_curve = <
-         /* ocv, charge, discharge */
-            3132      0       0
-            3273      0       0
-            3414      0       0
-            3555      0       0  
-            3625      1       3  
-            3660      2       7  
-            3696      7      14 
-            3731     12      22 
-            3766     18      37 
-            3801     37      49 
-            3836     51      57
-            3872     57      63
-            3942     69      73
-            4012     80      82
-            4083     89      90
-            4153    100     100>;
-	};
-
     pmu {
         compatible = "amlogic, aml_pmu_prober";                 /* fixed for pmu driver, don't change   */
         #address-cells = <1>;
diff --git a/arch/arm/mach-meson8b/clock.c b/arch/arm/mach-meson8b/clock.c
index d4a2f28279b6..a596f95568cf 100755
--- a/arch/arm/mach-meson8b/clock.c
+++ b/arch/arm/mach-meson8b/clock.c
@@ -60,7 +60,7 @@ static DEFINE_MUTEX(clock_ops_lock);
 #define SYS_PLL_TABLE_MIN	 24000000
 #define SYS_PLL_TABLE_MAX	2112000000
 
-#define CPU_FREQ_LIMIT 1488000000
+#define CPU_FREQ_LIMIT 1608000000
 
 struct sys_pll_s {
     unsigned int freq;
diff --git a/drivers/amlogic/power/aml_pmu/aml1218-sply.c b/drivers/amlogic/power/aml_pmu/aml1218-sply.c
index fa9faab083e1..4265aa1481a1 100755
--- a/drivers/amlogic/power/aml_pmu/aml1218-sply.c
+++ b/drivers/amlogic/power/aml_pmu/aml1218-sply.c
@@ -1221,8 +1221,8 @@ static int aml1218_update_state(struct aml_charger *charger)
     charger->vbat = aml1218_get_battery_voltage();
     charger->ocv  = aml1218_cal_ocv(charger->ibat, charger->vbat, charger->charge_status);
 
-    if (chg_status & 0x00004000) {
-        AML1218_DBG("%s, charge timeout, val:0x%02x, reset charger now\n", __func__, val);
+    if (chg_status & 0x0000400f) {
+        AML1218_DBG("%s, charge timeout or fault happen, status:0x%08x, reset charger now\n", __func__, chg_status);
         aml1218_set_charge_enable(0);
         msleep(1000);
         aml1218_set_charge_enable(1);
-- 
2.19.0

